src/core/content/connector.ts

```diff
@@ -279,15 +279,15 @@ export default class BaseConnector {
 	 *
 	 * @returns True if song is now playing; false otherwise
 	 */
-	public isPlaying: () => boolean;
+	public isPlaying: () => boolean | null | undefined;
 
 	/**
 	 * Default implementation to check whether a podcast is playing. Only has an
 	 * effect if the user has opted to disable podcast scrobbling.
 	 *
 	 * @returns True if the current track is a podcast; false otherwise
 	 */
-	public isPodcast: () => boolean = () => false;
+	public isPodcast: () => boolean | null = () => false;
```

Comment:
While I get the selector ones could potentially return undefined should we really be allowing these `is()` ones that seem boolean return anything other than boolean values? feels like we should strive to have a strict interface here to make our implementation a lot niceer

Reply:
The philosophy of web scrobbler v2 was that the connector functions all accepted undefined and null values in order to remove any checking functionality from connectors and put it into the core. This was not just an accident, but not doing null checks is explicitly stated as good practice in our wiki.

The core already handles these cases as v2 did (interpreting nullish value as false), and having a strict implementation vs a lenient implementation actually makes no difference to the core code itself, other than the declared type.

There are quite a few cases where these changes do remove a nullish coalescing operation from connector code (especially with isTrackArtDefault).

We could move to make it strict, but it would be a departure from the way the API has worked in the past and would probably just cause complexity without any big benefit except the core feeling cleaner.

Reply:
Gotcha - yeah it does make the connector implementation feel cleaner without having to be explicit.